crapssim/player.py

```diff
@@ -1,77 +1,114 @@
-class Player(object):
+import typing
+
+from crapssim.bet import Bet
+from crapssim.strategy import STRATEGY_TYPE, passline
+
+if typing.TYPE_CHECKING:
+    from crapssim.table import Table
+    from crapssim.dice import Dice
+
+
+class Player:
     """
     Player standing at the craps table
 
     Parameters
     ----------
-    bankroll : float
-        Starting amount of cash for the player, will be updated during play
+    bankroll : typing.SupportsFloat
+        Starting amount of cash for the player
     bet_strategy : function(table, player, unit=5)
         A function that implements a particular betting strategy.  See betting_strategies.py
-    name : string, optional (default = "Player")
+    name : string, default = "Player"
         Name of the player
+    unit : typing.SupportsFloat, default=5
+        Standard amount of bet to be used by bet_strategy
 
     Attributes
     ----------
+    bankroll : typing.SupportsFloat
+        Current amount of cash for the player
+    name : str
+        Name of the player
+    bet_strategy :
+        A function that implements a particular betting strategy. See betting_strategies.py.
+    strat_info : dict[str, typing.Any]
+        Variables to be used by the players bet_strategy
+    unit : typing.SupportsFloat
+        Standard amount of bet to be used by bet_strategy
     bets_on_table : list
         List of betting objects for the player
     total_bet_amount : int
         Sum of bet value for the player
     """
 
-    def __init__(self, bankroll, bet_strategy=None, name="Player"):
-        self.bankroll = bankroll
-        self.bet_strategy = bet_strategy
-        self.name = name
-        self.bets_on_table = []
-        self.total_bet_amount = 0
-        # TODO: initial betting strategy
+    def __init__(self, bankroll: typing.SupportsFloat,
+                 bet_strategy: STRATEGY_TYPE = passline,
+                 name: str = "Player",
+                 unit: typing.SupportsFloat = 5):
+        self.bankroll: float = bankroll
+        self.bet_strategy: STRATEGY_TYPE = bet_strategy
+        self.strat_info: dict[str, typing.Any] = {}
+        self.name: str = name
+        self.unit: typing.SupportsFloat = unit
+
+        self.bets_on_table: list[Bet] = []
+        self.total_bet_amount: float = 0.0
+
+    def bet(self, bet_object: Bet, table: 'Table') -> None:

```

Comment:
Thoughts on having a pointer to table within `self`, initialized when the player is added to the table?  Then we just reference `self.table` instead of having `table` as an argument?  

This would prevent the clunky mention of table every time and keep the syntax as is.

Reply:
I like that idea and added a new commit to facilitate that.

Reply:
Would we want to consider doing the same for Bet and have the table object as a variable on the Bet? It would allow us in the _update_bet method to remove the table_object and dice_object parameters.

Reply:
Since `_update_bet` is internal and only called once, I could go either way on it.  

In my mind, bets and players should only be on one table, so conceptually it makes sense to do a similar thing.